noise/README.md

```diff
@@ -221,6 +221,8 @@ syntax = "proto2";
 message NoiseExtensions {
     repeated bytes webtransport_certhashes = 1;
     repeated string stream_muxers = 2;
+    optional bool handshake_only = 3;
+    optional string tls_common_name = 4;
```

Comment:
Can we make this less specific to websocket/TLS? E.g., one way to do this is to:

1. Setup a noise stream.
2. Over the noise stream, send the TLS common name, etc.
3. Close the noise stream but leave the connection open.
4. Use the connection without encryption.

To support a better upgrade path, this method can be advertised as a new "stream muxer" (more like a "next protocol"). I.e.:

1. Initiator negotiates the "/drop-noise" protocol (or something like that) along with the expected peer ID (or nothing?).
2. Receiver returns their peer ID, and terminates their end of the noise stream.
3. Client terminates their end of the noise stream.

Reply:
This seems like it would add quite a lot of round trips?  Part of the joy of the extensions is they arrive as part of the handshake message.

I know connection establishment with WebSockets has a bunch of round trips already but is this a reason to make it slower, particularly when we already have WebTransport-specific stuff in the extensions registry?

Reply:
Hm. You're right, this'll add a round trip.